### PR TITLE
Add support for PrivateLink for AWS and Azure

### PR DESCRIPTION
**Pull request by tbroden84**
Adds support for PrivateLink for both AWS and Azure.

AWS:
* status
* allowed-principals
* service-name
* active-zones

Azure:
* status
* approved-subscriptions
* server-name
* alias

Make sure instance either have or can enable VPC before creating the PrivateLink.

### CODE REVIEW

**Comment by tbroden84** [comment]
Related to issues
https://github.com/cloudamqp/terraform-provider-cloudamqp/issues/142
https://github.com/cloudamqp/terraform-provider-cloudamqp/issues/154

File: cloudamqp/resource_cloudamqp_privatelink_aws.go [hunk 1]
```diff
@@ -63,6 +65,17 @@ func resourcePrivateLinkAws() *schema.Resource {
 				Description: "Configurable timeout in seconds when enable PrivateLink",
 			},
 		},
+		CustomizeDiff: customdiff.All(
+			customdiff.ValidateValue("allowed_principals", func(value, meta interface{}) error {
```
Comment (tbroden84) [inline cloudamqp/resource_cloudamqp_privatelink_aws.go:69]: Worked with the defined custom diff `ValidateValue`.

File: cloudamqp/resource_cloudamqp_privatelink_azure.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+	d.SetId(fmt.Sprintf("%d", instanceID))
+	params["approved_subscriptions"] = d.Get("approved_subscriptions").([]interface{})
+	if len(params) > 0 {
+		err := api.UpdatePrivatelink(instanceID, params)
+		if err != nil {
+			return err
+		}
+	}
+
+	return resourcePrivateLinkAwsRead(d, meta)
```
Comment (dentarg) [inline cloudamqp/resource_cloudamqp_privatelink_azure.go:101]: I think this should be:

```suggestion
	return resourcePrivateLinkAzureRead(d, meta)
```

Comment (tbroden84) [reply cloudamqp/resource_cloudamqp_privatelink_azure.go:101]: True

**Pull request merged**